### PR TITLE
[REVIEW] Adding templates for Issues and Pull Requests

### PR DESCRIPTION
**Pull request by mike-wendt**
Borrowed from [Numba](https://github.com/numba/numba/tree/master/.github) these templates will help to communicate the process used by the GoAi team to the larger community.

### CODE REVIEW

**Comment by mike-wendt** [comment]
Once approved here I will push the same to `libgdf`

File: .github/ISSUE_TEMPLATE.md [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+## Reporting a bug
+
+<!--
+
+Before submitting a bug report please ensure that you can check off these boxes:
+
+-->
+
+- [ ] I am using the latest version of PyGDF from conda or built from `master`.
```
Comment (kkraus14) [inline .github/ISSUE_TEMPLATE.md]: I'd really like to capture dependency information as well if possible. Things like CUDA, Pandas, Numpy, Arrow versions that have the potential to break things.

Comment (mike-wendt) [reply .github/ISSUE_TEMPLATE.md]: Agreed that would be very helpful. I added in two boxes to fill in environment and version information, let me know if you think I have it covered now.

**Comment by scopatz** [comment]
LGTM!

**Comment by kkraus14** [comment]
Looks good to me, thanks @mike-wendt!

**Pull request merged**